ecc/enclave_chaincode.go

```diff
@@ -61,13 +61,21 @@ func (t *EnclaveChaincode) Invoke(stub shim.ChaincodeStubInterface) pb.Response
 	function, _ := stub.GetFunctionAndParameters()
 	logger.Debugf("Invoke is running [%s]", function)
 
-	if function == "setup" { // create enclave and register at ercc
+	// Look first if there are system functions (and handle them)
+	if function == "__setup" { // create enclave and register at ercc
+		// Note: above string is also used in ecc_valdation_logic.go,
+		// so if you change here you also will have to change there ...
+		// If/when we refactor we should define such stuff somewhere as constants..
 		return t.setup(stub)
-	} else if function == "getEnclavePk" { //get Enclave PK
+	} else if function == "__getEnclavePk" { //get Enclave PK
 		return t.getEnclavePk(stub)
-	} else {
-		return t.invoke(stub)
+	} else if function == "__init" { // call FPC's init in enclave
```

Comment:
why is fpc "__init" invoked per "fabric invoke" instead of "fabric init"?

Reply:
not sure i understand: `init` is implicitly called during instantiate but the go-chaincode `init` is of course too early to do anything in the enclave? So we have no choice other than call it through `invoke`?  (ecc of course would then forward the `__init` payload to the enclave's `init`

Reply:
do we need `__init` at this point, or should it rather be our `setup`?
In the former case, I would drop it. In the latter case, I would rename `setup`

Reply:
@Bruno: Ok, init is now eradicated from the PR (overall took me almost same time as implementing init e2e).   (see https://github.com/hyperledger-labs/fabric-private-chaincode/compare/49e97fe1791720a48915236f3dde544fcd765156..519c4db2d4b6d589d00aa3c791baa12f74c26c2e).

Following Marcus' dictum this morning , i've changed the assignee to you.

As Ann is waiting for this PR, would be good to get reviews in soon